modules/xmpp/RtxModifier.js

```diff
@@ -0,0 +1,276 @@
+import { getLogger } from "jitsi-meet-logger";
+const logger = getLogger(__filename);
+import * as transform from 'sdp-transform';
+const SDPUtil = require("./SDPUtil");
+
+/**
+ * Begin helper functions
+ */
+/**
+ * Given a videoMLine, returns a list of the video
+ *  ssrcs (those used to actually send video, not
+ *  any associated secondary streams)
+ * @param {object} videoMLine media line object from transform.parse
+ * @returns {list<string>} list of primary video ssrcs
+ */
+function getPrimaryVideoSsrcs (videoMLine) {
+    let videoSsrcs = videoMLine.ssrcs
+        .map(ssrcInfo => ssrcInfo.id)
+        .filter((ssrc, index, array) => array.indexOf(ssrc) === index);
+
+    if (videoMLine.ssrcGroups) {
+        videoMLine.ssrcGroups.forEach((ssrcGroupInfo) => {
+            // Right now, FID groups are the only ones we parse to 
+            //  disqualify streams.  If/when others arise we'll
+            //  need to add support for them here
+            if (ssrcGroupInfo.semantics === "FID") {
+                // secondary FID streams should be filtered out
+                let secondarySsrc = ssrcGroupInfo.ssrcs.split(" ")[1];
+                videoSsrcs.splice(
+                  videoSsrcs.indexOf(parseInt(secondarySsrc)), 1);
+            }
+        });
+    }
+    return videoSsrcs;
+}
+
+/**
+ * Given a video mline (as parsed from transform.parse),
+ *  and a primary ssrc, return the corresponding rtx ssrc
+ *  (if there is one) for that video ssrc
+ * @param {object} videoMLine the video MLine from which to extract the
+ *  rtx video ssrc
+ * @param {number} primarySsrc the video ssrc for which to find the
+ *  corresponding rtx ssrc
+ * @returns {number} the rtx ssrc (or undefined if there isn't one)
+ */
+function getRtxSsrc (videoMLine, primarySsrc) {
+    if (videoMLine.ssrcGroups) {
+        let fidGroup = videoMLine.ssrcGroups.find(group => {
+            if (group.semantics === "FID") {
+                let groupPrimarySsrc = SDPUtil.parseGroupSsrcs(group)[0];
+                return groupPrimarySsrc === primarySsrc;
+            }
```

Comment:
Also you can save one `SDPUtil.parseGroupSsrcs` call if you store the result from it.